Content.Server/Fluids/EntitySystems/SpillableSystem.cs

```diff
@@ -125,12 +130,34 @@ private void AddSpillVerb(EntityUid uid, SpillableComponent component, GetVerbsE
         Verb verb = new();
         verb.Text = Loc.GetString("spill-target-verb-get-data-text");
         // TODO VERB ICONS spill icon? pouring out a glass/beaker?
-        verb.Act = () =>
+        if (component.SpillDelay == null)
         {
-            var puddleSolution = _solutionContainerSystem.SplitSolution(args.Target,
-                solution, solution.DrainAvailable);
-            SpillAt(puddleSolution, Transform(args.Target).Coordinates, "PuddleSmear");
-        };
+            verb.Act = () =>
+            {
+                var puddleSolution = _solutionContainerSystem.SplitSolution(args.Target,
+                    solution, solution.DrainAvailable);
+                SpillAt(puddleSolution, Transform(args.Target).Coordinates, "PuddleSmear");
+            };
+        }
+        else
+        {
+            verb.Act = () =>
+            {
+                if (component.CancelToken == null)
+                {
+                    component.CancelToken = new CancellationTokenSource();
+                    _doAfterSystem.DoAfter(new DoAfterEventArgs(args.User, component.SpillDelay.Value, component.CancelToken.Token, component.Owner)
+                    {
+                        BreakOnTargetMove = true,
+                        BreakOnUserMove = true,
+                        BreakOnDamage = true,
+                        BreakOnStun = true,
```

Comment:
Should it need a hand? The doafter kind of implies you aren't just kicking it over

Reply:
That's logical. I'll add that